xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/internal/fileupload/FileUploadUtils.java

```diff
@@ -144,4 +153,43 @@ public FileItem createItem(String fieldName, String contentType, boolean isFormF
             }
         }
     }
+
+    private static class PartAttachmentValidationSupplier implements AttachmentValidationSupplier
```

Comment:
Really not a fan of this: sounds like something that we might want to be able to reuse, and not only there...